grazer/src/main/java/edu/asu/diging/grazer/core/fileupload/service/impl/FileUploadServiceImpl.java

```diff
@@ -0,0 +1,105 @@
+
+@Service
+@PropertySource(value = "classpath:config.properties")
+public class FileUploadServiceImpl implements IFileUploadService {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+    
+    @Autowired
+    private FileMetadataDatabaseConnection connection;
+
+    @Autowired
+    private Environment env;
+    
+    public synchronized String createID() {
+        return UUID.randomUUID().toString();
+    } 
+    
+    private String createDirectory() {
+        File directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        while(directory.exists()) {
+            directory = new File(env.getProperty("transformation.file.dir") + File.separator + createID());
+        }
+        directory.mkdir();
+        return directory.toString();
+    }
+    
+    private String uploadFiles(CommonsMultipartFile[] multipartFiles) throws IOException {
+
+        String directory = createDirectory();
+
+        for(CommonsMultipartFile file: multipartFiles) {
+            try {      
+                String serverFileName = directory + File.separator + file.getOriginalFilename();
+                File serverFile = new File(serverFileName);
+
+                BufferedOutputStream stream = new BufferedOutputStream(new FileOutputStream(serverFile));
+                stream.write(file.getBytes());
+                stream.close();
+            } catch(IOException e) {
+                logger.error("You failed to upload " + file.getOriginalFilename(), e);
+                throw new IOException(e);
+            } 
+        } 
+        return directory;
+    }
+
+    @Override
+    public void save(FileUploadFormImpl transformationMetadataAndFiles, Principal principal) throws IOException { 
+        
+        TransformationFilesImpl transformation = new TransformationFilesImpl();
+        TransformationFilesMetadataImpl metadata = new TransformationFilesMetadataImpl();
+        
+        transformation.setTransformationFileContent(transformationMetadataAndFiles.getTransformationFile().getBytes());
+        transformation.setPatternFileContent(transformationMetadataAndFiles.getPatternFile().getBytes());
+        transformation.setTransformationFileName(transformationMetadataAndFiles.getTransformationFile().getOriginalFilename());
+        transformation.setPatternFileName(transformationMetadataAndFiles.getPatternFile().getOriginalFilename());
+        
+        metadata.setLabel(transformationMetadataAndFiles.getLabel());
+        metadata.setDescription(transformationMetadataAndFiles.getDescription());
+        metadata.setDate(OffsetDateTime.now());
+        metadata.setUploader(principal.getName());
+        
+        CommonsMultipartFile[] files = new CommonsMultipartFile[2];
+        files[0] = transformationMetadataAndFiles.getTransformationFile();
+        files[1] = transformationMetadataAndFiles.getPatternFile();
+
+        metadata.setPath(uploadFiles(files));
```

Comment:
what if the folder with all the transformation file is moved somewhere else? (e.g. because the hard drive is full)